experimenter/experimenter/experiments/changelog_utils.py

```diff
@@ -64,16 +64,32 @@ def generate_nimbus_changelog(experiment, changed_by, message):
             "published_dto"
         ) != experiment_data.get("published_dto")
 
-    return NimbusChangeLog.objects.create(
-        experiment=experiment,
-        old_status=old_status,
-        old_status_next=old_status_next,
-        old_publish_status=old_publish_status,
-        new_status=experiment.status,
-        new_status_next=experiment.status_next,
-        new_publish_status=experiment.publish_status,
-        changed_by=changed_by,
-        experiment_data=experiment_data,
-        published_dto_changed=published_dto_changed,
-        message=message,
-    )
+    if changed_on:
+        return NimbusChangeLog.objects.create(
+            experiment=experiment,
+            old_status=old_status,
+            old_status_next=old_status_next,
+            old_publish_status=old_publish_status,
+            new_status=experiment.status,
+            new_status_next=experiment.status_next,
+            new_publish_status=experiment.publish_status,
+            changed_by=changed_by,
+            experiment_data=experiment_data,
+            published_dto_changed=published_dto_changed,
+            message=message,
+            changed_on=changed_on,
+        )
+    else:
+        return NimbusChangeLog.objects.create(
+            experiment=experiment,
+            old_status=old_status,
+            old_status_next=old_status_next,
+            old_publish_status=old_publish_status,
+            new_status=experiment.status,
+            new_status_next=experiment.status_next,
+            new_publish_status=experiment.publish_status,
+            changed_by=changed_by,
+            experiment_data=experiment_data,
+            published_dto_changed=published_dto_changed,
+            message=message,
+        )
```

Comment:
why if else?

Reply:
The `changed_on` parameter was just added for a single test and I didn't want to break the existing test cases. Please do let me know if there was a better way of handling this and I'll update it accordingly.

Reply:
You shouldn't need the if-else since the value is `None`.